Controller/ConfigController.php

```diff
@@ -186,30 +235,38 @@ public function deleteAction(Request $request, $id)
             return $this->redirect($this->generateUrl('integrated_channel_config_index')); // data is already gone
         }
 
+        $event = new GetResponseConfigEvent($data, $request);
+
+        if ($this->dispatcher->dispatch(IntegratedChannelEvents::CONFIG_DELETE_REQUEST, $event)->getResponse()) {
+            return $event->getResponse();
+        }
+
         $form = $this->createDeleteForm($data);
+        $form->handleRequest($request);
+
+        if ($form->get('actions')->getData() == 'cancel') {
+            return $this->redirect($this->generateUrl('integrated_channel_config_index'));
+        }
 
-        if ($request->isMethod('delete')) {
-            $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->manager->remove($data);
 
-            if ($form->get('actions')->getData() == 'cancel') {
-                return $this->redirect($this->generateUrl('integrated_channel_config_index'));
+            if ($message = $this->getFlashMessage()) {
+                $message->success(sprintf('The config %s is removed', $data->getName()));
             }
 
-            if ($form->isValid()) {
-                $this->manager->remove($data);
+            $response = $this->redirect($this->generateUrl('integrated_channel_config_index'));
 
-                if ($message = $this->getFlashMessage()) {
-                    $message->success(sprintf('The config %s is removed', $data->getName()));
-                }
+            $this->dispatcher->dispatch(
+                IntegratedChannelEvents::CONFIG_DELETE_RESPONSE,
+                new FilterResponseConfigEvent($data, $request, $response)
+            );
 
-                return $this->redirect($this->generateUrl('integrated_channel_config_index'));
-            }
+            return $response;
         }
 
         return $this->render('IntegratedChannelBundle:Config:delete.html.twig', [
-            'adapter' => $this->registry->hasAdapter(
-                $data->getAdapter()
-            ) ? $this->registry->getAdapter($data->getAdapter()) : null,
+            'adapter' => $this->registry->hasAdapter($data->getAdapter()) ? $this->registry->getAdapter($data->getAdapter()) : null,
```

Comment:
This makes Jenkins fails: http://jenkins.e-active.nl:8080/job/integrated-channel-bundle/35/console